pandas/_libs/groupby.pyx

```diff
@@ -154,7 +155,7 @@ def group_cumprod_float64(float64_t[:, :] out,
         int64_t lab
 
     N, K = (<object>values).shape
-    accum = np.ones_like(values)
+    accum = np.ones((ngroups, K), np.float64)
```

Comment:
Not sure this is the best approach - why would we size `accum` to be `ngroups x K` here but then iterate over `N x K`?

Reply:
because we index into accum with label[i], which ranges from 0 to ngroups-1?